File: tests/fixtures/python/vulnerable_jwt.py
"""Fixture: JWT signing with quantum-vulnerable asymmetric algorithms (PQC011)."""

import jwt
from jose import jwt as jose_jwt


def issue_rs256(payload: dict, private_key) -> str:
    # PQC011 — RS256 (RSA signature) JWT.
    return jwt.encode(payload, private_key, algorithm="RS256")


def issue_es256(payload: dict, private_key) -> str:
    # PQC011 — ES256 (ECDSA signature) JWT.
    return jwt.encode(payload, private_key, algorithm="ES256")


def issue_ps256(payload: dict, private_key) -> str:
    # PQC011 — PS256 (RSA-PSS signature) JWT.
    return jwt.encode(payload, private_key, algorithm="PS256")


def verify_rs256(token: str, public_key):
    # PQC011 — decoding restricted to an asymmetric algorithm.
    return jwt.decode(token, public_key, algorithms=["RS256"])


def jose_rs256(payload: dict, key) -> str:
    # PQC011 — python-jose with an RSA algorithm.
    return jose_jwt.encode(payload, key, algorithm="RS256")

File: tests/fixtures/javascript/vulnerable_jwt.js
// Fixture: jsonwebtoken with quantum-vulnerable asymmetric algorithms (PQC011).
const jwt = require('jsonwebtoken');

function issueRs256(payload, privateKey) {
  // PQC011 — RS256 (RSA signature).
  return jwt.sign(payload, privateKey, { algorithm: 'RS256' });
}

function issueEs256(payload, privateKey) {
  // PQC011 — ES256 (ECDSA signature).
  return jwt.sign(payload, privateKey, { algorithm: 'ES256' });
}

function verify(token, publicKey) {
  // PQC011 — verification restricted to an asymmetric algorithm.
  return jwt.verify(token, publicKey, { algorithms: ['RS256'] });
}

const { SignJWT, jwtVerify } = require('jose');

async function issueJose(claims, privateKey) {
  // PQC011 — jose: asymmetric algorithm in the protected header.
  return new SignJWT(claims).setProtectedHeader({ alg: 'RS256' }).sign(privateKey);
}

async function verifyJose(token, publicKey) {
  // PQC011 — jose: verification restricted to an asymmetric algorithm.
  return jwtVerify(token, publicKey, { algorithms: ['ES384'] });
}

module.exports = { issueRs256, issueEs256, verify, issueJose, verifyJose };
